ocrd_utils/ocrd_utils/config.py

```diff
@@ -0,0 +1,167 @@
+        except KeyError as e:
+            if var_obj.has_default:
+                raw_value = var_obj.default() if callable(var_obj.default) else var_obj.default
+            else:
+                raise e
+        if not var_obj.validator(raw_value):
+            raise ValueError(f"'{name}' set to invalid value '{raw_value}'")
+        return var_obj.parser(raw_value)
+
+    def is_set(self, name):
+        if not name in self._variables:
+            raise ValueError(f"Unregistered env variable {name}")
+        return name in environ
+
+    def raw_value(self, name):
+        if not name in self._variables:
+            raise ValueError(f"Unregistered env variable {name}")
+        return environ[name]
+
+config = OcrdEnvConfig()
+
+config.add('OCRD_METS_CACHING',
+    description='If set to `true`, access to the METS file is cached, speeding in-memory search and modification.',
+    validator=lambda val: val in ('true', 'false', '0', '1'),
+    parser=lambda val: val in ('true', '1'))
+
+config.add('OCRD_MAX_PROCESSOR_CACHE',
+    description="Maximum number of processor instances (for each set of parameters) to be kept in memory (including loaded models) for processing workers or processor servers.",
+    parser=int,
+    default=(True, 128))
+
+config.add("OCRD_PROFILE",
+    description="""\
+Whether to enable gathering runtime statistics
+on the `ocrd.profile` logger (comma-separated):
+- `CPU`: yields CPU and wall-time,
+- `RSS`: also yields peak memory (resident set size)
+- `PSS`: also yields peak memory (proportional set size)
+""",
+  validator=lambda val : all(t in ('', 'CPU', 'RSS', 'PSS') for t in val.split(',')),
+  default=(True, ''))
+
+config.add("OCRD_PROFILE_FILE",
+    description="If set, then the CPU profile is written to this file for later peruse with a analysis tools like snakeviz")
+
+config.add("OCRD_DOWNLOAD_RETRIES",
+    description="Number of times to retry failed attempts for downloads of workspace files.",
+    validator=int,
+    parser=int)
+
+def _ocrd_download_timeout_parser(val):
+    timeout = val.split(',')
+    if len(timeout) > 1:
+        timeout = tuple(float(x) for x in timeout)
+    else:
+        timeout = float(timeout[0])
+    return timeout
+
+config.add("OCRD_DOWNLOAD_TIMEOUT",
+    description="Timeout in seconds for connecting or reading (comma-separated) when downloading.",
+    parser=_ocrd_download_timeout_parser)
+
+config.add("OCRD_NETWORK_SERVER_ADDR_PROCESSING",
+        description="Default address of Processing Server to connect to (for `ocrd network client processing`).",
+        default=(True, ''))
+
+config.add("OCRD_NETWORK_SERVER_ADDR_WORKFLOW",
+        description="Default address of Workflow Server to connect to (for `ocrd network client processing`).",
+        default=(True, ''))
+
+config.add("OCRD_NETWORK_SERVER_ADDR_WORKSPACE",
+        description="Default address of Workspace Server to connect to (for `ocrd network client processing`).",
```

Comment:
Should be `ocrd network client workspace`?

Reply:
Fixed in 1e35a5074